lxd/certificates.go

```diff
@@ -377,6 +380,20 @@ func clusterMemberJoinTokenValid(d *Daemon, r *http.Request, projectName string,
 			return nil, fmt.Errorf("Failed to cancel operation %q: %w", foundOp.ID, err)
 		}
 
+		var expiry time.Time
+
+		// Depending on whether it's a local operation or not, expiry will either be a time.Time or a string.
+		if d.serverName == foundOp.Location {
+			expiry, _ = foundOp.Metadata["expiresAt"].(time.Time)
+		} else {
+			expiry, _ = time.Parse(time.RFC3339Nano, foundOp.Metadata["expiresAt"].(string))
+		}
+
+		// Check if token has expired.
+		if time.Now().After(expiry) {
+			return nil, errExpiredToken
```

Comment:
You can avoid the need for errExpiredToken by doing:

```
return nil, api.StatusErrorf(http.StatusForbidden, "Token has expired")
```

And then where the err of `clusterMemberJoinTokenValid` is used:

```go
if err != nil {
    return response.SmartError(fmt.Errorf("Failed during search for join token operation: %w", err))
}
```

Which will then use the right HTTP status code.